### DISCUSSION TITLE
Matter provisioning and integration for new Tasmota devices without WiFi credentials

### DISCUSSION THREAD

**Discussion by ivanvach** [arendst/Tasmota#20648] [Ideas]
For the sake of example say "Martin Jerry" brand manufacturers ESP32 based light switches with Tasmota pre-flashed. Is it possible for them to print QR code on box and for the user to adopt the device by Matter enabled Google Home (or other) environment.

This is exactly what happens with the adoption of Sonoff MINIR4M switch ( ESP32 based) that is the first Sonoff device that is Matter enabled.  However Sonoff MINIR4M is not a Tasmota device.

The question is can Tasmota make something similar to what Sonoff did with their proprietary firmware?

**Comment by sfromis** [comment]
If you "only" want Matter, you may not get any value out of having Tasmota on the device. Tasmota is more meant to provide added value as it can provide a lot of flexibility with MQTT control, and home automation. Also on-device automations for much more than just the rather rudimentary Matter features.

Tasmota is designed for being on a Wifi (or Ethernet) network, with Matter lately coming as an extra feature on top of what Tasmota already can provide. Operating as a standalone end device on Matter would take some adaptation, and may not be such a good Tasmota use case anyway.

I suppose that a project could be based on Tasmota, to provide things like sensor access, or control of special devices, and then coming to act as an end device on Matter, but that's another thing than having a QR code for Matter-only enrollment.

**Comment by ivanvach** [comment]
Well, looks like I'm brainwashed that Matter is the future of IoT. With this said when Google and Apple put their weight behind something we better watch. 

I do agree that Tasmota is ahead of the game as far as versatility goes, but who is using this versatility. Not the average Joe Doe out there who wants it quick and easy.

**Reply by sfromis** [reply]
Well, the thing is that Tasmota is not really intended for "the average Joe Doe", but for users with "the willingness to tinker and learn", including researching and solving potential problems. In this context, supporting Matter is an extra thing to tinker with, not the sole purpose of having anything Tasmota. **If** Matter becomes a success (not a given), it would be kinda annoying to tinkerers not  having this feature in their box of toys.

Tasmota is not a good fit for users liable to go "help me, my device does not work" in case if any trouble, and expecting product support plus robustness against user errors.

**Reply by ivanvach** [reply]
I disagree about the last statement. First Tasmota is quite solid now better than some other "rush to market" half baked commercial products. Second I received timely and better quality help here than for many other "commercial" products. Keep up with the good work Tasmota team !
